department-of-veterans-affairs/abd-vro
# BIP Claims Evidence Final Integration

## What was the problem?


- Claims Evidence API was not fully integrated.

Associated tickets or Slack threads:

- [MCP-2009](https://amida.atlassian.net/browse/MCP-2009)

## How does this fix it?[^1]


This PR finalizes the Claims Evidence API integration

- Payload and Header fields in call the BIP CE finalized. We are using FILENUMBER for the E-Folder folder. Payload fIeld values discussed in slack discussions updated.
- Services for BIP API and BIP Evidence API are separated. This will enable to select in service mocks for one without effecting other.
- No temporary file is save anymore. Byte[] is directly send.
- Also moved the primary bean RestTemplate. RestTemplate for CE API was causing problems

## How to test this PR
There is not an easy way to test to locally since the Mock Claims Evidence API is not fully integrated to docker compose yet and otherwise BIP API's are not locally availabkle. But the following what I will do. If it is too much just look at the code.
- Make sure application builds and tests out of the box in local.
- Set ENV=dev. Three other environment variables should be set. BIP_PASSWORD to keystore_pw, BIP_KEYSTORE to value in https://github.com/department-of-veterans-affairs/abd-vro/blob/develop/mock-bip-ce-api/src/test/resources/client-keystore.yml, BIP_TRUSTSTORE to value in https://github.com/department-of-veterans-affairs/abd-vro/blob/develop/mock-bip-ce-api/src/test/resources/truststore.yml.
- Remove Service and conditional from MockBIPCEService. Remove conditional from BIPCEService.
- Build the application,
- Run the mock service in a new terminal: ./gradlew :mock-bip-ce-api:bootRun
- Run the positive MAS end2end test or exercise the MAS end point with the collection Id there.
- Look at the terminal. It should have recieved the payload and file. The file will be save to a temp file. Take a look at the file to see if it is the expected PDF.


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
